### PR TITLE
fix: date comparison without timezone in consideration

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Even after reading new posts, the icon from the reading streak does not update to be filled.
- The value of `lastViewAt` is in UTC when fetched from the API, which does not match the local time. With the update, we will utilize the user's set timezone. We can also use the current local timezone but it won't match other stats since behind the scenes all we use is the set timezone from the profile.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://mi-620.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/streak/ReadingStreakButton.tsx [hunk 1]
```diff
@@ -71,11 +87,11 @@ export function ReadingStreakButton({
   className,
 }: ReadingStreakButtonProps): ReactElement {
   const { logEvent } = useLogContext();
+  const { user } = useAuthContext();
   const isLaptop = useViewSize(ViewSize.Laptop);
   const isMobile = useViewSize(ViewSize.MobileL);
   const [shouldShowStreaks, setShouldShowStreaks] = useState(false);
-  const hasReadToday =
-    new Date(streak?.lastViewAt).getDate() === new Date().getDate();
+  const hasReadToday = getHasReadToday(streak?.lastViewAt, user?.timezone);
```
Comment (rebelchris) [inline packages/shared/src/components/streak/ReadingStreakButton.tsx:94]: But shouldn't API just return it in the correct format already?

**Review by rebelchris — COMMENTED** [review]
More of a question why can't we handle this in API returning the value we expect?

File: packages/shared/src/components/streak/ReadingStreakButton.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/streak/ReadingStreakButton.tsx:94]: Currently, what is returned is the stored value, which is UTC.

Comment (sshanzel) [reply packages/shared/src/components/streak/ReadingStreakButton.tsx:94]: We can maybe add a prop where the returned value involves the right tz.

Comment (rebelchris) [reply packages/shared/src/components/streak/ReadingStreakButton.tsx]: So then we don't need this one?

**Review by rebelchris — APPROVED** [review]

File: packages/shared/src/components/streak/ReadingStreakButton.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/streak/ReadingStreakButton.tsx]: For backwards compatibility, I have kept the original column, and introduced a new one with timezone considered. So we'd still need this.

**Pull request merged**